Server/src/test/java/Deck_Test.java

```diff
@@ -0,0 +1,150 @@
+package test.java;
+
+import org.junit.Assert;
+import org.junit.Test;
+import server.dataCenter.models.card.Deck;
+import server.dataCenter.models.card.ServerCard;
+import server.exceptions.ClientException;
+import server.exceptions.LogicException;
+import shared.models.card.AttackType;
+import shared.models.card.CardType;
+
+import java.util.ArrayList;
+
+public class Deck_Test {
+
+    private ServerCard makeMinionCard() {
+        return new ServerCard(" ", " ", " ", " ", CardType.MINION, new ArrayList<>(), 1, 1, 1, 1, AttackType.MELEE, 1);
+    }
+
+    private ServerCard makeGeneralCard() {
+        return new ServerCard(" ", " ", " ", " ", CardType.HERO, new ArrayList<>(), 1, 1, 1, 1, AttackType.MELEE, 1);
+    }
+
+    @Test
+    public void hasCard_whenMissingGeneral_NotFindGeneral() {
+        Deck sut = new Deck("sut");
+        ServerCard general = makeGeneralCard();
+
+        boolean actualHasCard = sut.hasCard(general.getCardId());
+        boolean expectedHasCard = false;
+
+        Assert.assertEquals(expectedHasCard, actualHasCard);
+    }
+
+    @Test
+    public void hasCard_whenMissingCard_NotFindCard() throws LogicException {
+        Deck sut = new Deck("sut");
+        ServerCard minion = makeMinionCard();
+        sut.addCard(minion);
+
+        boolean actualHasCard = sut.hasCard(minion.getCardId());
+        boolean expectedHasCard = true;
+
+        Assert.assertEquals(expectedHasCard, actualHasCard);
+    }
+
+    @Test
+    public void hasCard_whenHasGeneral_FindGeneral() throws LogicException {
+        Deck sut = new Deck("sut");
+        ServerCard minion = makeGeneralCard();
+        sut.addCard(minion);
+
+        boolean actualHasCard = sut.hasCard(minion.getCardId());
+        boolean expectedHasCard = true;
+
+        Assert.assertEquals(expectedHasCard, actualHasCard);
+    }
+
+    @Test
+    public void hasCard_whenHasCard_FindCard() throws LogicException {
+
+        Deck sut = new Deck("sut");
+        ServerCard minion = makeMinionCard();
+        sut.addCard(minion);
+
+        boolean actualHasCard = sut.hasCard(minion.getCardId());
+        boolean expectedHasCard = true;
+
+        Assert.assertEquals(expectedHasCard, actualHasCard);
+    }
+
+    @Test(expected = ClientException.class)
+    public void addCard_whenAddSecondGeneral_throwsClientException() throws LogicException {
+        Deck sut = new Deck("sut");
+        ServerCard general0 = makeGeneralCard();
+        general0.setCardId("0");
+        ServerCard general1 = makeGeneralCard();
+        general1.setCardId("1");
+        sut.addCard(general0);
+        sut.addCard(general1);
+    }
+
+    @Test
+    public void removeCard_whenHasCard_removesCard() throws LogicException {
+        Deck sut = new Deck("sut");
+        ServerCard minion = makeMinionCard();
+        sut.addCard(minion);
+        sut.removeCard(minion);
```

Comment:
add assert after add to check card actually is added

Reply:
I don't think so. Good to have one assertion per test, and checking that a card is added is already done.

Reply:
As a general rule, fewer asserts is better.  If you have another test checking add then I'm fine omitting the 2nd assert.